### ISSUE TITLE
Sparse SDPA sink

### ISSUE THREAD

**Issue by jmitrovicTT** [tenstorrent/tt-metal#54173]
## Ask

Add an optional `attention_sink` argument to `ttnn.transformer.sparse_sdpa`, with the same semantics
the dense `scaled_dot_product_attention` already has. Needed for DeepSeek-V4 CSA prefill.

## What the sink is

A learned vector, **one scalar per attention head** — nothing per-token, nothing data-dependent.
`[64]` for V4-Flash, `[128]` for V4-Pro
(`models/demos/deepseek_v3_d_p/reference/deepseek_v4/modeling_deepseek_v4.py:797`).

Mathematically it is **one extra term in the softmax denominator**:

```
N[c]  =  Σ_j exp( s[j] ) · v[j, c]        numerator   — unchanged
D     =  Σ_j exp( s[j] )                  denominator

without sink:   out[c]  =  N[c] /  D
with sink:      out[c]  =  N[c] / ( D + exp σ[h] )
```

The reference writes it as a concatenated logit column that gets dropped after softmax
([modeling_deepseek_v4.py](https://github.com/tenstorrent/tt-metal/blob/main/models/demos/deepseek_v3_d_p/reference/deepseek_v4/modeling_deepseek_v4.py):730-743), but that is equivalent to the above

## It is already implemented in the dense kernel

[ttnn/cpp/ttnn/operations/transformer/sdpa/device/kernels/compute/compute_common.hpp](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/transformer/sdpa/device/kernels/compute/compute_common.hpp)
